src/components/WorkspaceStarControl.ts

```diff
@@ -0,0 +1,99 @@
+import _ from 'lodash/fp';
+import { FC, useState } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { Clickable } from 'src/components/common';
+import { icon, spinner } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import colors from 'src/libs/colors';
+import { withErrorReporting } from 'src/libs/error';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { useStore } from 'src/libs/react-utils';
+import { AuthState, authStore } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+interface WorkspaceStarControlProps {
+  workspace: WorkspaceWrapper;
+}
+
+export const WorkspaceStarControl: FC<WorkspaceStarControlProps> = ({ workspace }) => {
```

Comment:
This style of type signature makes components more consistent with other functions.

```suggestion
export const WorkspaceStarControl = ({ workspace }: WorkspaceStarControlProps): ReactNode => {
```

Same comment for other components.